src/main/java/com/welovecoding/nbeditorconfig/processor/EditorConfigProcessor.java

```diff
@@ -118,7 +118,9 @@ protected FileInfo excuteOperations(DataObject dataObject, MappedEditorConfig co
     StringBuilder content;
 
     try {
-      content = new StringBuilder(primaryFile.asText());
+      MappedCharset mc = config.getCharset();
+      String charset = mc != null ? mc.getCharset().name() : StandardCharsets.UTF_8.name();
```

Comment:
I think that the FileEncodingQuery should be used if the MappedCharset is null. What do you think? If I may change to it, I'll add a new commit.